cumulusci/tasks/salesforce/CreateCommunity.py

```diff
@@ -0,0 +1,84 @@
+import json
+import requests
+from datetime import datetime
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+from cumulusci.core.exceptions import SalesforceException
+
+
+class CreateCommunity(BaseSalesforceApiTask):
+    api_version = "46.0"
+    task_docs = """
+    Create a Salesforce Community via the Connect API.
+    Specify the `template` "VF Template" for Visualforce Tabs community,
+    or the name for a specific desired template
+    """
+    task_options = {
+        "template": {
+            "description": "Name of the template for the community.",
+            "required": True,
+        },
+        "name": {"description": "Name of the community.", "required": True},
+        "description": {
+            "description": "Description of the community.",
+            "required": False,
+        },
+        "url_path_prefix": {
+            "description": "URL prefix for the community.",
+            "required": True,
+        },
+        "timeout": {
+            "description": "Time to wait, in seconds, for the community to be created",
+            "default": 120,
+        },
+    }
+
+    def _init_options(self, kwargs):
+        super(CreateCommunity, self)._init_options(kwargs)
+        self.options["timeout"] = int(self.options.get("timeout", 120))
+
+    def _run_task(self):
+        self.logger.info('Creating community "{}"'.format(self.options["name"]))
+        payload = {
+            "name": self.options["name"],
+            "description": self.options.get("description") or "",
+            "templateName": self.options["template"],
+            "urlPathPrefix": self.options["url_path_prefix"],
+        }
+
+        # Before we can create a Community, we have to click through the "New Community"
+        # button in the All Communities setup page. (This does some unknown behind-the-scenes setup).
+        # Let's simulate that without actually using a browser.
+        self.logger.info("Preparing org for Communities")
+        s = requests.Session()
+        s.get(self.org_config.start_url)
```

Comment:
We should probably get the response here and call response.raise_for_status() so that we'll error out if authentication fails.

Reply:
👍